### ISSUE TITLE
IntelliJ finds wrong struct when there are multiple with cfg attributes

### ISSUE THREAD

**Issue by LoganDark** [intellij-rust/intellij-rust#5712]
## Environment

* **IntelliJ Rust plugin version:** 0.2.106.2135-192
* **Rust toolchain version:** 1.44.1
* **IDE name and version:** IntelliJ IDEA Ultimate 2019.2
* **Operating system:** macOS Mojave 10.14.6

## Problem description
When multiple structs with the same name exist, both with `cfg` attributes, IDEA tends to find the wrong one, or nothing at all

## Steps to reproduce

On a macOS machine, try to use [`fruitbasket`](https://crates.io/crates/fruitbasket)`::Trampoline::new`.

It doesn't autocomplete, IDEA doesn't know its function definition, and the unknown-ness is completely infective. Everything you do with the Result is just an `<unknown>`. Inspections stop working on the unknown types. In fact if you try to find the definition it actually takes you to the _wrong_ definition, proving that IntelliJ is finding _something_, just the wrong something.

The `new` function exists in all cases, and is not conditional. While IntelliJ does derp things, the compiler knows everything that is going on.

Here is a code example which shows a very related issue:

```rust
#[cfg(target_os = "macos")]
struct Something {}

#[cfg(target_os = "macos")]
impl Something {
	fn on_macos(&self) {
		println!("On macOS!")
	}
}

#[cfg(not(target_os = "macos"))]
struct Something {}

#[cfg(not(target_os = "macos"))]
impl Something {
	fn off_macos(&self) {
		println!("Off macOS!")
	}
}

fn main() {
	let something = Something {};
	something.on_macos();
}
```

IntelliJ doesn't autocomplete either one of the methods and in fact tries to tell you that it's an error because of a duplicate definition. Both of these are wrong and should be corrected. Finding the definition of the Something within `main` doesn't pick one, but shows you both instead, so it's not the exact same issue.

It is honestly sickening that these issues have been sitting around for years with little to no support - see #1191, #5458, etc.

**Comment by Undin** [comment]
@LoganDark compiler cfg options were supported a long time ago (see #4537)
But you are using too old plugin version which doesn't contain the corresponding changes.
With latest plugin versions everything is fine

![2020-07-09 14 51 32](https://user-images.githubusercontent.com/2539310/87036577-b076a280-c1f3-11ea-83a0-e04e5b7dc5e7.gif)

Consider upgrading your IDE and plugin.
Note, the latest version of the plugin is available only for 2020.1 and 2020.2

**Comment by LoganDark** [comment]
> Note, the latest version of the plugin is available only for 2020.1 and 2020.2

That's disappointing, not everyone is using bleeding edge releases

**Comment by Undin** [comment]
Sorry, but it's rather hard to maintain old platform versions because of incompatibilities between them.
We guarantee to support only the latest stable major platform version and the latest EAPs.

**Comment by LoganDark** [comment]
It takes a while to update, and then I have to resolve all the incompatibilities with old plugins that worked just fine before the update. That's why many people with a productive workflow like myself don't feel like updating the moment a new release comes out. The fact that you can't even maintain support for 2019.2 is just sad.

Since I do need Rust support I'm forced to update which causes all sorts of headaches for me and my plugins.

I don't want to rant about how stupid it is to only support the very latest version. I'm sure you all know about this since you're developers on IntelliJ's platform as well, but the disregard you have for others' desire to maintain a stable installation is simply shocking.

**Comment by Undin** [comment]
@LoganDark BTW initial support of compiler cfg options was introduced in #4232 and should be available in `0.2.106.2135-192`. Bit it's disabled by default. You can try to enable it via `rust.cfg.attributes.enabled` registry option. It's quite hard to guarantee that everything will work but you can try. Note, Cargo features are not supported in this version anyway.

**Comment by LoganDark** [comment]
Thank you, at least IDEA discovers the fact that the methods exist. It still finds the wrong one so documentation doesn't work, but I'm reading the online documentation anyway so this is fine.

Thanks for your help. I'm still a bit disappointed you consider a version of the IDE released less than a year ago "old", but it's too late to fix that now.
